### ISSUE TITLE
Laravel 5.4 v1.2

### ISSUE THREAD

**Issue by marcelogarbin** [Arcesilas/active-state#1]
Hi,

I'm currently with Laravel 5.4, with "pyaesone17/active-state" v1.2. 

1. When switching to your package, is the issue of dynamic URLs serviced for this version of Laravel 5.4(https://github.com/Arcesilas/active-state#route-checks)?
2. What do I need to take note of to make this exchange?

Good update work.

Thank's

**Comment by Arcesilas** [comment]
Hi,

Thanks for your interest.
I'm afraid I don't understand what you mean... English is not my mother tongue and I prefer to be sure to understand what you mean to give you an answer.

Please note that this package requires Laravel/Framework 5.5, since it uses features that were not present before (and not only PHP 7 syntaxes). I may try to backport it for Laravel 5.4, but I cannot give any guaranty.

**Comment by marcelogarbin** [comment]
Ok, I'm using google translator hehehe. My language is Portuguese (pt_BR). What is your language?

So I'm finalizing a project in Laravel 5.4 and would like to use that part of dynamic URLs only(Route Checks). If there is not for L5.4, no problem.

I will try to start the next projects in Laravel 5.5, I just have not migrated yet because I use a package (https://nwidart.com/laravel-modules/v1/introduction) that is not yet 100% for L5.5.

When I start in L5.5 I will use your updates for this package, I can also help you with future improvements if you need to.

Thank's for your attention =)

**Comment by Arcesilas** [comment]
I'm French and that's what I love with internet: so many cultures meeting here :)

Thanks for giving details about your case and what you want to achieve!

With little work, you may add the checkRouteIs() feature:
1. Extend `Pyaesone17\Active\Active` class with one of your own, and place it in the namespace of your choice.

2. Add a function `checkRouteParamsBoolean()` in your own Active class, with the code of `checkRouteIs()` from this package (just copy/paste, it should work):
```php
    public function checkRouteParamsBoolean($route, array $routeParameters = [])
    {
        if (!is_array($route)) {
            $route = [$route => $routeParameters];
        }
        foreach ($route as $r => $parameters) {
            try {
                $routeUri = route($r, $parameters, true);
                if ($this->request->fullUrlIs($routeUri)) {
                    return true;
                }
            } catch (\Exception $e) {
                // Just ignore the exception: route does not exist
            }
        }
        return false;
    }
```

and a checkRouteParams() method:
```php
    public function checkRouteParams($route, array $parameters = [], $active = null, $inactive = null)
    {
        $this->setReturnValue($active,$inactive);
        return $this->checkRouteParamsBoolean($route, $parameters) ? $this->activeValue : $this->inActiveValue;
    }
```

3. Extend `Pyaesone17\Active\ActiveStateServiceProvider`. Unregister the default one and replace it with your own.
In your own ActiveStateServiceProvider, override the register method to make it register your own ActiveState class:
```php
    public function register()
    {
        $this->app->singleton('active-state', function ($app) {
            return new \App\ActiveState\Active($app->request);
        });
        $this->mergeConfigFrom(
            __DIR__.'/config/active.php', 'active'
        );
    }
```

Then, you can use something like:
Active::checkRouteParams('category.show', ['slug' => $category->slug]);

*I've deliberately named the function checkRouteParams to avoid confusion with this package. Feel free to use the name of your choice, obviously.*

If you want a Blade directive, you can add it in your ServiceProvider. Just extend the boot method and add the directive:

```php
    public function boot()
    {
        parent::boot();
        Blade::directive('ifActiveRouteParams'), function ($expression) {
            return '<?php if(Active::checkRouteParamsBoolean($expression)): ?>';
        }

        Blade::directive('endIfActiveRouteParams'), function ($expression) {
            return '<?php if(Active::checkRouteParamsBoolean($expression)): ?>';
        }
    }
```

This code has been completely and entirely not tested :) You may need to fix some bug, typo or whatever mistake I could have made.

**Comment by marcelogarbin** [comment]
I had to make some changes to work, I could not use the `__DIR__` the path was the one of the class and not the one of the project, I used `base_path('config/active.php')`:

```
class ActiveStateServiceProvider extends \Pyaesone17\ActiveState\ActiveStateServiceProvider
{
    public function boot()
    {
        parent::boot();

        Blade::directive('ifActiveRouteParams', function ($expression) {
            return '<?php if(Active::checkRouteParamsBoolean($expression)): ?>';
        });

        Blade::directive('endIfActiveRouteParams', function ($expression) {
            return '<?php endif; ?>';
        });
    }

    public function register()
    {
        $this->app->singleton('active-state', function ($app) {
            return new \App\ExtendsClass\Active($app->request);
        });

        $this->mergeConfigFrom(
            base_path('config/active.php'), 'active'
        );
    }
}
```

I had never done this (extend class and register a new provider by extending another), everything worked as expected!

Merci pour l'aide mon ami. :smile:

**Comment by Arcesilas** [comment]
Great! Glad I could help!
And thanks again for your interest in this package (event if you don't really use it for now :)

**Comment by Arcesilas** [comment]
Hi,

Just to let you know I've released a new version 1.2.1 of this package, based on the version 1.2.0 by Pyaesone17, which add the functionnality you needed. Basically, I've integrated this hack into the 1.2.0 version.

You may try it (I have not written any test for 1.x yet) and let me know if you experience any problem. Using this version of the package let you benefit from support and future updates, since it seems Pyaesone17 has not much time right now to agree to development (that's why I've forked the repository and make it live on its own).

To use it, I guess you may use:

`composer require ^1.2`

Since you use L5.4 and version 2.0 requires L5.5, you may stick with the appri=opriate version (and branch 1.x will stick to L5.4).
